packages/experimental-app-router/src/client.ts

```diff
@@ -31,12 +35,28 @@ export const { getClient } = registerApolloClient(() => {
     },
   };
 
-  const linkChain = createHttpLink({
+  let linkChain = createHttpLink({
     uri: getGraphqlEndpoint(),
   });
 
   // @todo Create hook for client and options.
 
+  // If the request is coming from the auth client, apply the auth link.
+  if (authenticated) {
+    linkChain = setContext(async (_, { headers }) => {
+      // get the authentication token from local storage if it exists
+      const token = await fetchAccessToken();
```

Comment:
Fetch the access token whenever we call `getAuthClient`. With the caching system in place in Next.js 13 and App Router, this request can be cached, and have that cache evicted before the `accessToken` expires, since we receive the expiration time as well.